pkg/apis/v1alpha1/suite_test.go

```diff
@@ -115,4 +120,96 @@ var _ = Describe("Validation", func() {
 			Expect(ant.Validate(ctx)).To(Not(Succeed()))
 		})
 	})
+	var _ = Describe("AWSNodeTemplate Hash", func() {
+		var awsnodetemplatespec v1alpha1.AWSNodeTemplateSpec
+		var awsnodetemplate *v1alpha1.AWSNodeTemplate
+		BeforeEach(func() {
+			awsnodetemplatespec = v1alpha1.AWSNodeTemplateSpec{
+				AWS: v1alpha1.AWS{
+					InstanceProfile: aws.String("profile-1"),
+					Tags: map[string]string{
+						"keyTag-1": "valueTag-1",
+						"keyTag-2": "valueTag-2",
+					},
+					LaunchTemplate: v1alpha1.LaunchTemplate{
+						MetadataOptions: &v1alpha1.MetadataOptions{
+							HTTPEndpoint: aws.String("test-metadata-1"),
+						},
+						BlockDeviceMappings: []*v1alpha1.BlockDeviceMapping{
+							{
+								DeviceName: aws.String("map-device-1"),
+							},
+							{
+								DeviceName: aws.String("map-device-2"),
+							},
+						},
+					},
+				},
+				UserData:           aws.String("userdata-test-1"),
+				DetailedMonitoring: aws.Bool(false),
+			}
+			awsnodetemplate = test.AWSNodeTemplate(awsnodetemplatespec)
+		})
+		It("should change hash when static fields are updated", func() {
+			expectedHash := awsnodetemplate.Hash()
+
+			// Change one static field for 5 provisioners
```

Comment:
nit: technically 6, but should be fine to do like this:
```suggestion
			// Create an AWSNodeTemplate for each static field
```